drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -610,6 +760,29 @@ local function handle_set_color_temperature(driver, device, cmd)
   device:send(req)
 end
 
+local function handle_valve_open(driver, device, cmd)
+  local endpoint_id = device:component_to_endpoint(cmd.component)
+  local req = clusters.ValveConfigurationAndControl.server.commands.Open(device, endpoint_id)
+  device:send(req)
+end
+
+local function handle_valve_close(driver, device, cmd)
+  local endpoint_id = device:component_to_endpoint(cmd.component)
+  local req = clusters.ValveConfigurationAndControl.server.commands.Close(device, endpoint_id)
+  device:send(req)
+end
+
+local function handle_set_level(driver, device, cmd)
+  local commands = clusters.ValveConfigurationAndControl.server.commands
+  local endpoint_id = device:component_to_endpoint(cmd.component)
+  local level = cmd.args.level
+  if level ~= 0 then
+    device:send(commands.Open(device, endpoint_id, nil, level))
+  else
+    device:send(commands.Close(device, endpoint_id))
+  end
```

Comment:
Out of curiosity @nickolas-deboom why do we have this logic here? Does 0 not work correctly?

Reply:
I think this was a leftover from earlier testing... I switched the order in the latest commit though. I also added new test cases to verify the setLevel command.